app/backend/src/couchers/jobs/worker.py

```diff
@@ -0,0 +1,147 @@
+
+        try:
+            ret = func(message_type.FromString(job.payload))
+            job.state = BackgroundJobState.completed
+            logger.info(f"Job #{job.id} complete on try number {job.try_count}")
+        except Exception as e:
+            logger.exception(e)
+            if job.try_count >= job.max_tries:
+                # if we already tried max_tries times, it's permanently failed
+                job.state = BackgroundJobState.failed
+                logger.info(f"Job #{job.id} failed on try number {job.try_count}")
+            else:
+                job.state = BackgroundJobState.error
+                # exponential backoff
+                job.next_attempt_after += timedelta(seconds=15 * (2 ** job.try_count))
+                logger.info(f"Job #{job.id} error on try number {job.try_count}, next try at {job.next_attempt_after}")
+            # add some info for debugging
+            job.failure_info = traceback.format_exc()
+
+        # exiting ctx manager commits and releases the row lock
+    return True
+
+
+def service_jobs():
+    """
+    Service jobs in an infinite loop
+    """
+    # multiprocessing uses fork() which in turn copies file descriptors, so the engine may have connections in its pool
+    # that we don't want to reuse. This is the SQLALchemy-recommended way of clearing the connection pool in this thread
+    get_engine().dispose()
+
+    while True:
+        # if no job was found, sleep for a second, otherwise query for another job straight away
+        if not process_job():
+            sleep(1)
+
+
+def _run_job_and_schedule(sched, schedule_id):
+    job_type, frequency = SCHEDULE[schedule_id]
+    logger.info(f"Processing job of type {job_type}")
+
+    # wake ourselves up after frequency
+    sched.enter(
+        frequency.total_seconds(),
+        1,
+        _run_job_and_schedule,
+        argument=(
+            sched,
+            schedule_id,
+        ),
+    )
+
+    # queue the job
+    queue_job(job_type, empty_pb2.Empty())
+
+
+def run_scheduler():
+    """
+    Schedules jobs according to schedule in .definitions
+    """
+    # multiprocessing uses fork() which in turn copies file descriptors, so the engine may have connections in its pool
+    # that we don't want to reuse. This is the SQLALchemy-recommended way of clearing the connection pool in this thread
+    get_engine().dispose()
+
+    sched = scheduler(monotonic, sleep)
+
+    for schedule_id, (job_type, frequency) in enumerate(SCHEDULE):
+        sched.enter(
+            0,
+            1,
+            _run_job_and_schedule,
+            argument=(
+                sched,
+                schedule_id,
+            ),
+        )
+
+    sched.run()
+
+
+def _run_forever(func):
+    while True:
+        try:
+            logger.critical("Background worker starting")
+            func()
+        except Exception as e:
+            logger.critical("Unhandled exception in background worker", exc_info=e)
```

Comment:
Perhaps a grace period here to not consume 100% cpu and keep hammering the database in case of a programming error. Remove the loop or put a `time.sleep(60)` here.